File: 백준/Silver/1380. 귀걸이/귀걸이.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int cnt = 1;
        
        while (true) {
            int n = Integer.parseInt(br.readLine().trim());
            if (n == 0) break;

            List<String> names = new ArrayList<>();
            Set<Integer> goods = new HashSet<>();

            for (int i = 0; i < n; i++) {
                names.add(br.readLine().trim());
            }

            for (int i = 0; i < 2 * n - 1; i++) {
                String[] input = br.readLine().split(" ");
                int num = Integer.parseInt(input[0]);
                
                if (goods.contains(num)) {
                    goods.remove(num);
                } else {
                    goods.add(num);
                }
            }

            int remaining = goods.iterator().next(); 
            System.out.println(cnt + " " + names.get(remaining - 1)); 

            cnt++;
        }
    }
}

File: 백준/Silver/1380. 귀걸이/귀걸이.py
import sys
input = sys.stdin.readline

cnt = 1
while True:
    name = []
    goods = set()
    n = int(input().strip())
    if n == 0:
        break
    else:
        for i in range(n):
            name.append(input().strip())

        for i in range(2*n-1):
            num, type = input().split()
            num = int(num)
            if num in goods:
                goods.remove(num)
            else:
                goods.add(num)

    print(cnt, name[goods.pop()-1])  
    cnt += 1
